cmd/gardener-metrics-exporter/app.go

```diff
@@ -122,30 +125,37 @@ func run(o *options, closeCh chan os.Signal) error {
 	return nil
 }
 
-func setupInformerFactories(kubeconfigPath string, stopCh <-chan struct{}) (gardeninformers.SharedInformerFactory, kubeinformers.SharedInformerFactory, error) {
-	k8sGardenClient, err := kubernetes.NewClientFromFile(kubeconfigPath)
+func newConfigFromBytes(kubeconfig string) *restclient.Config {
+	kubecf, err := ioutil.ReadFile(kubeconfig)
 	if err != nil {
-		return nil, nil, err
+		fmt.Println("Could not read config")
+		os.Exit(1)
 	}
+	configObj, err := clientcmd.Load(kubecf)
```

Comment:
`Load` returns a pointer to a config object which can potentially be nil. We should catch this case.